save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DemoRunComponent.kt

```diff
@@ -181,52 +196,54 @@ private fun diktatDemoComponent() = FC<DiktatDemoComponentProps> { props ->
                     "custom-select"
                 ) { event ->
-                    setDiktatRunRequest { runRequest ->
+                    setDemoRunRequest { runRequest ->
                         runRequest.copy(
                             mode = DemoMode.valueOf(event.target.value)
                         )
                     }
                 }
             }
         }
-        div {
-            className = ClassName("mb-1 mt-1 d-flex justify-content-center")
-            label {
-                className = ClassName("btn btn-outline-secondary m-0")
-                val reader = FileReader().apply {
-                    onload = { event ->
-                        setDiktatRunRequest { runRequest ->
-                            (event.target.asDynamic()["result"] as String?)?.let {
-                                runRequest.copy(
-                                    config = it.split("\n")
-                                )
-                            } ?: runRequest
+        props.configName?.let { configName ->
+            div {
+                className = ClassName("mb-1 mt-1 d-flex justify-content-center")
+                label {
+                    className = ClassName("btn btn-outline-secondary m-0")
+                    val reader = FileReader().apply {
+                        onload = { event ->
+                            setDemoRunRequest { runRequest ->
+                                (event.target.asDynamic()["result"] as String?)?.let {
+                                    runRequest.copy(
+                                        config = it.split("\n")
+                                    )
+                                } ?: runRequest
+                            }
                         }
                     }
-                }
-                input {
-                    type = InputType.file
-                    multiple = false
-                    hidden = true
-                    onChange = { event ->
-                        event.target.files!!.asList()
-                            .firstOrNull()
-                            ?.let { file ->
-                                reader.readAsText(file, "UTF-8")
-                            }
+                    input {
+                        type = InputType.file
+                        multiple = false
+                        hidden = true
+                        onChange = { event ->
+                            event.target.files!!.asList()
+                                .firstOrNull()
+                                ?.let { file ->
+                                    reader.readAsText(file, "UTF-8")
+                                }
+                        }
                     }
+                    fontAwesomeIcon(icon = faUpload)
+                    val uploadOrReplace = if (demoRunRequest.config.isNullOrEmpty()) {
+                        "Upload"
+                    } else {
+                        "Replace"
```

Comment:
This doesn't make sense for now

Reply:
it doesn't make sense for Diktat only, `com.saveourtool.save.demo.DemoRunRequest` supports `config` == null

Reply:
Well, I mean I am not sure if the label works correctly. Seems that with diktat it didn't work as there was default config in data class. For now it should work fine probably